engine/runtime/src/main/java/org/enso/interpreter/node/controlflow/caseexpr/StringLiteralBranchNode.java

```diff
@@ -39,9 +42,26 @@ void doText(
     }
   }
 
+  @Specialization(
+      guards = {"targetInterop.isString(target)"},
+      limit = "3")
+  void doInteropString(
+      VirtualFrame frame,
+      Object state,
+      Object target,
+      @CachedLibrary("target") InteropLibrary targetInterop) {
+    try {
+      if (textProfile.profile(equalStrings(literal, targetInterop.asString(target)))) {
+        accept(frame, state, new Object[0]);
+      }
+    } catch (UnsupportedMessageException e) {
+      throw CompilerDirectives.shouldNotReachHere(e);
+    }
+  }
+
   @CompilerDirectives.TruffleBoundary
```

Comment:
`equalStrings` is behind `@TruffleBoundary`, that cannot be fast.

Reply:
This has not been changed in this PR at all.

I only made us use the single entry-point `Core_Text_Utils.equals` that should be used for _all_ string equality in Enso (and is used almost everywhere). It still delegates to exactly the same `Normalizer.compare` as before, but now we use the same entry point, so in case we ever change the equality logic we only have to modify one place.